### DISCUSSION TITLE
Proper way to set value of select option trait programmatically

### DISCUSSION THREAD

**Discussion by saysujan** [GrapesJS/grapesjs#4366] [Q&A]
Hi, I'm trying to set a default value for custom component on component:add where I've to perform certain logic there, I tried to set the value but neither it triggers change:object_types nor it shows selected option in trait panel. Any help is appreciated. https://jsfiddle.net/3rwn1a9p/125/

**Comment by artf** [comment]
I'd probably suggest performing the logic in the component `init` method as it sounds more reasonable, but in the case of the `component:add` you should probably do something like this:
```js
editor.on("component:add", (comp) => {
  if (comp.is('my-custom-component')) {
    // Update component prop/attribute in order to reflect it in traits
    comp.set('object_types', '1')
  }
});
```
